edge/pkg/edgehub/process.go

```diff
@@ -128,10 +130,44 @@ func (eh *EdgeHub) routeToEdge() {
 	}
 }
 
+var (
+	msgStoreLock sync.Mutex
+	msgStore     map[string][]*model.Message
+)
+
 func (eh *EdgeHub) sendToCloud(message model.Message) error {
-	eh.keeperLock.Lock()
-	err := eh.chClient.Send(message)
-	eh.keeperLock.Unlock()
+	var (
+		err             error
+		resource        = message.GetResource()
+		operation       = message.GetOperation()
+		isQuery         = (operation == model.QueryOperation)
+		_, restype, _   = metamanager.ParseResource(resource)
+		isConfigMap     = restype == "configmap"
+		needSendSyncMsg = true
+	)
+
+	if isQuery && isConfigMap {
```

Comment:
Thanks for the fixing, except configmap, if query other objects, we may also need to add code here, so do we need to find a general solution for it?

Reply:
I remove the flag `isConfigMap`.